examples/jsm/loaders/TaskManager.js

```diff
@@ -0,0 +1,560 @@
+/**
+ * @author Don McCurdy / https://www.donmccurdy.com
+ * @author Kai Salmen / https://kaisalmen.de
+ */
+
+import { FileLoaderBufferAsync } from "./obj2/utils/FileLoaderBufferAsync.js";
+
+/**
+ *
+ */
+class TaskManager {
+
+    constructor () {
+
+        this.types = new Map();
+        this.verbose = false;
+
+    }
+
+    /**
+     * Set if logging should be verbose
+     *
+     * @param {boolean} verbose
+     * @return {TaskManager}
+     */
+    setVerbose ( verbose ) {
+
+        this.verbose = verbose;
+        return this;
+
+    }
+
+    /**
+     * Returns true if support for the given task type is available.
+     * @param {string} type The type as string
+     * @return boolean
+     */
+    supportsType ( type ) {
+
+        return this.types.has( type );
+
+    }
+
+    /**
+     * Registers functionality for a new task type.
+     * @param {string} type The name to be used for registration.
+     * @param {number} maximumWorkerCount How many workers are allows. Set 0 for execution on Main ({@link FakeTaskWorker})
+     * @param {function} initFunction The function to be called when the worker is initialised
+     * @param {function} executeFunction The function to be called when the worker is executed
+     * @param {function} comRoutingFunction The function that should handle communication, leave undefined for default behavior
+     * @param {String[]} [dependencyUrls]
+     * @return {TaskManager}
+     */
+    registerType ( type, maximumWorkerCount, initFunction, executeFunction, comRoutingFunction, dependencyUrls ) {
+
+        let workerTypeDefinition = new WorkerTypeDefinition( type, maximumWorkerCount, this.verbose );
+        workerTypeDefinition.setFunctions( initFunction, executeFunction, comRoutingFunction );
+        workerTypeDefinition.setDependencyUrls( dependencyUrls );
+        this.types.set( type, workerTypeDefinition );
+        return this;
+
+    }
+
+    /**
+     * Registers functionality for a new task type based on module file.
+     * @param {string} type The name to be used for registration.
+     * @param {number} maximumWorkerCount How many workers are allows. Set 0 for execution on Main ({@link FakeTaskWorker})
+     * @param {string} workerJsmUrl The URL to be used for the Worker. Module must provide logic to handle "init" and "execute" messages.
+     * @return {TaskManager}
+     */
+    registerTypeJsm ( type, maximumWorkerCount, workerJsmUrl ) {
```

Comment:
I've never seen the term "jsm" anywhere outside the name of this folder, so I'd prefer to use a clearer term here. It's also important that even though the "examples/js" folder is being removed later this year, that has very little to do with whether this "jsm" method should be used — DRACOLoader, BasisTextureLoader, and KTX2Loader will all use the non-JSM register method. What do you think of:

```js
// Default.
manager.registerTask( type, maxCount, initFn, execFn, ... );

// URL. (undecided which name..)
manager.registerTaskModule( type, maxCount, moduleUrl );
manager.registerTaskUrl( ... );
manager.registerTaskSrc( ... );
```

I think the fact that the script happens to be an ES Module is less important than the fact that it's an external script. That means any user with a build system like webpack or browserify will need to specifically configure their application to output files in the right directories.

Reply:
Agreed, let's remove jsm from method signatures. `module` is the best replacement term as it is also in line with the naming of the `Worker` constructor parameter, I think.

Reply:
Changed in latest commit.